### PR TITLE
Init

### PR DESCRIPTION
**Pull request by ajlee21**
This PR initializes the ponyo repo using the functions established in https://github.com/ajlee21/simulate-expression-compendia/tree/master/functions

All of these functions have been previously reviewed in past PRs. 

This PR is pull those functions out into its own repo, in order to create a pypi module to be installed on my analysis repos (simulate-expression-compendia and generic-expression-patterns) and hopefully users interested in doing an analysis similar to my analysis repos.

This PR contains the pulled functions and a setup file. Next PR will add tests and dependency files.

### CODE REVIEW

File: ponyo/generate_template_data.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
+'''
+Author: Alexandra Lee
+Date Created: 15 June 2020
+
+Script to generate gene expression compendia using experiment-preserving approach using a specific template experiment.
```
Comment (dongbohu) [inline ponyo/generate_template_data.py:5]: See my comments on docstring.

File: ponyo/generate_template_data.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+import ast
+import pandas as pd
+import numpy as np
+import random
+import glob
+import pickle
+from keras.models import load_model
+from sklearn import preprocessing
+
+from ponyo import generate_labeled_data
```
Comment (dongbohu) [inline ponyo/generate_template_data.py:19]: Wouldn't `import generate_labeled_data` work?

File: ponyo/generate_template_data.py [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+    loaded_decode_model.load_weights(weights_decoder_file)
+
+    # Read data
+    normalized_data = pd.read_table(
+        normalized_data_file,
+        header=0,
+        sep='\t',
+        index_col=0).T
+
+    #print("Normalized gene expression data contains {} samples and {} genes".format(
```
Comment (dongbohu) [inline ponyo/generate_template_data.py:117]: Are these two lines for debug only?

File: ponyo/generate_template_data.py [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+    simulated_data_encoded_df = data_encoded_df.apply(
+        lambda x: x + shift_vec_df, axis=1)
+
+    # Decode simulated data into raw gene space
+    simulated_data_decoded = loaded_decode_model.predict_on_batch(
+        simulated_data_encoded_df)
+
+    simulated_data_decoded_df = pd.DataFrame(simulated_data_decoded,
+                                                index=simulated_data_encoded_df.index,
```
Comment (dongbohu) [inline ponyo/generate_template_data.py:167]: Alignment not correct

File: ponyo/generate_template_data.py [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+                                                columns=selected_data_df.columns)
```
Comment (dongbohu) [inline ponyo/generate_template_data.py:168]: Alignment not correct.

File: ponyo/generate_template_data.py [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+    
+    simulated_data_scaled = scaler.inverse_transform(simulated_data_decoded_df)
+
+    simulated_data_scaled_df = pd.DataFrame(simulated_data_scaled,
+                                            columns=simulated_data_decoded_df.columns,
+                                            index=simulated_data_decoded_df.index)
+
+    #print("Return: simulated gene expression data containing {} samples and {} genes".format(
```
Comment (dongbohu) [inline ponyo/generate_template_data.py:176]: Debugging lines?

File: ponyo/generate_template_data.py [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+    #    simulated_data_scaled_df.shape[0], simulated_data_scaled_df.shape[1]))
+
+    # Save
+    out_file = os.path.join(local_dir,
+    "pseudo_experiment",
+    "selected_simulated_data_"+selected_experiment_id+"_"+str(run)+".txt")
+
+    simulated_data_scaled_df.to_csv(
+        out_file, float_format='%.3f', sep='\t')
```
Comment (dongbohu) [inline ponyo/generate_template_data.py:185]: New line char needed.

File: ponyo/pipeline.py [hunk 8]
```diff
@@ -0,0 +1,470 @@
...
+import gc
+from sklearn import preprocessing
+
+from joblib import Parallel, delayed
+import multiprocessing
+
+import warnings
+warnings.filterwarnings(action='ignore')
+
+from ponyo import vae, utils, simulations
```
Comment (dongbohu) [inline ponyo/pipeline.py:28]: `from ponyo` is redundant.

File: ponyo/similarity_metric_parallel.py [hunk 9]
```diff
@@ -0,0 +1,270 @@
...
+import ast
+import pandas as pd
+import numpy as np
+import random
+import glob
+import warnings
+warnings.filterwarnings(action='ignore')
+
+sys.path.append("../")
+from ponyo import cca_core
```
Comment (dongbohu) [inline ponyo/similarity_metric_parallel.py:19]: `from ponyo` seems redundant.

File: ponyo/simulations.py [hunk 10]
```diff
@@ -0,0 +1,304 @@
...
+import glob
+import pickle
+import pandas as pd
+import numpy as np
+import gc
+
+import warnings
+warnings.filterwarnings(action='ignore')
+
+from ponyo import generate_data_parallel
```
Comment (dongbohu) [inline ponyo/simulations.py:19]: `from ponyo` is redundant on these two lines.

File: ponyo/vae.py [hunk 11]
```diff
@@ -0,0 +1,331 @@
...
+import numpy as np
+import random as rn
+
+from keras.layers import Input, Dense, Lambda, Layer, Activation
+from keras.layers.normalization import BatchNormalization
+from keras.models import Model, Sequential
+from keras import metrics, optimizers
+from keras.callbacks import Callback
+
+from ponyo.helper_vae import sampling_maker, CustomVariationalLayer, WarmUpCallback
```
Comment (dongbohu) [inline ponyo/vae.py:23]: `from ponyo.helper_vae` to `from helper_vae`?

File: setup.py [hunk 12]
```diff
@@ -0,0 +1,11 @@
+from setuptools import setup
+
+setup(name='ponyo',
+version='0.1',
```
Comment (dongbohu) [inline setup.py:4]: You may want to add indentation on lines 4-11, since they are all arguments to `setup()` function.

File: setup.py [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+description='Install functions from ponyo: simulate gene expression compendia',
+url='#',
+author='Alexandra Lee',
+author_email='alexjlee.21@gmail.com',
+license='BSD 3-Clause',
+packages=['ponyo'],
+zip_safe=False)
```
Comment (dongbohu) [inline setup.py:11]: New line char needed.

File: ponyo/helper_vae.py [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+'''
+
+import os
+import argparse
+import pandas as pd
+import matplotlib.pyplot as plt
+import tensorflow as tf
+
+# To ensure reproducibility using Keras during development
+# https://keras.io/getting-started/faq/#how-can-i-obtain-reproducible-results-using-keras-during-development
```
Comment (ben-heil) [inline ponyo/helper_vae.py:15]: You use the logic from this link in your code, but the comment itself is separated from the code it refers to

File: ponyo/pipeline.py [hunk 15]
```diff
@@ -0,0 +1,470 @@
...
+                            intermediate_dim,
+                            latent_dim,
+                            epsilon_std,
+                            normalized_data,
+                            base_dir,
+                            dataset_name,
+                            train_architecture)
+
+    del normalized_data, params
+    gc. collect()
```
Comment (ben-heil) [inline ponyo/pipeline.py:297]: Space is not needed between `gc.` and `collect()`

File: ponyo/similarity_metric_parallel.py [hunk 16]
```diff
@@ -0,0 +1,270 @@
...
+import sys
+import ast
+import pandas as pd
+import numpy as np
+import random
+import glob
+import warnings
+warnings.filterwarnings(action='ignore')
+
+sys.path.append("../")
```
Comment (ben-heil) [inline ponyo/similarity_metric_parallel.py:18]: You shouldn't need to mess with the pythonpath; cca_core is in the same directory as similarity_metric_paralell

File: ponyo/simulations.py [hunk 17]
```diff
@@ -0,0 +1,304 @@
...
+                                                                         dataset_name,
+                                                                         analysis_name)
+
+  # Convert similarity scores to pandas dataframe
+  similarity_score_df = pd.DataFrame(data={'score': batch_scores},
+                                     index=lst_num_partitions,
+                                     columns=['score'])
+
+  similarity_score_df.index.name = 'number of partitions'
+  similarity_score_df
```
Comment (ben-heil) [inline ponyo/simulations.py:301]: This looks like a debugging statement, so can probably be removed

File: ponyo/generate_template_data.py (hunk 2 above)
Comment (ajlee21) [reply ponyo/generate_template_data.py:19]: Technically i believe I can, this was just to be explicit

**Comment by ajlee21** [comment]
Thank you for being so thorough! I believe I addressed all the comments. Let me know if it looks ok to merge

**Comment by dongbohu** [comment]
A few super-trivial issues. After you address them, please feel free to close it. Thanks.

**Pull request merged**